fmriprep/config.py

```diff
@@ -401,14 +404,16 @@ def init(cls):
         if cls._layout is None:
             import re
             from bids.layout import BIDSLayout
-            work_dir = cls.work_dir / 'bids.db'
-            work_dir.mkdir(exist_ok=True, parents=True)
+            _db_path = cls.database_path or (cls.work_dir / cls.run_uuid / 'bids.db')
+            _db_path.mkdir(exist_ok=True, parents=True)
             cls._layout = BIDSLayout(
                 str(cls.bids_dir),
                 validate=False,
-                # database_path=str(work_dir),
+                database_path=_db_path,
+                reset_database=cls.database_path is None,
                 ignore=("code", "stimuli", "sourcedata", "models",
                         "derivatives", re.compile(r'^\.')))
+            cls.database_path = _db_path
```

Comment:
I think this implies reusing the cache if you're reusing the config?

Reply:
I added this since the config is saved and reloaded during workflow generation
https://github.com/poldracklab/fmriprep/blob/3b1dabdb3b5205ed48771a9fa0d9638f61c80d36/fmriprep/cli/run.py#L29-L49
and I wanted to ensure it wouldn't be re-indexed on reload.

Good point, we will have to differentiate between intra/inter-run config reuse - though I think that is something to worry about in the context of the reuse PR.